src/peerconnection.cpp

```diff
@@ -146,11 +146,21 @@ void PeerConnection::setLocalDescription(Description::Type type) {
 	impl()->changeSignalingState(newSignalingState);
 	signalingLock.unlock();
 
-	if (impl()->gatheringState == GatheringState::New) {
+	if (impl()->gatheringState == GatheringState::New && !impl()->config.disableAutoGathering) {
 		iceTransport->gatherLocalCandidates(impl()->localBundleMid());
 	}
 }
 
+void PeerConnection::gatherLocalCandidates(std::vector<IceServer> additionalIceServers) {
+	auto iceTransport = impl()->initIceTransport();
```

Comment:
Since the ICE transport is created here, why not simply add the additional ICE servers to the config just before? Then you wouldn't need any custom logic in `IceTransport`.

Reply:
Sorry, I was mislead by the call here, this should be `getIceTransport()` and you should throw a `logic_error` if null, as it means no local description has been set.